### PR TITLE
Refactors relationships to allow implement target scrpts

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java [hunk 1]
```diff
@@ -628,46 +593,37 @@ Operation getOperation() {
...
         }
 
         @Override
         Object getScript(ImplementationArtifact artifact, Operation op) {
-            Optional<RelationshipTemplate> optionalRelationshipTemplate = findRelationshipRequirement(nodeId, toscaApplication, requirementId);
+            Optional<RelationshipTemplate> optionalRelationshipTemplate = Optional.fromNullable(toscaApplication.getNodeTemplate(nodeId).getRelationships().get(relationship.getRelationshipId()));
             if (!optionalRelationshipTemplate.isPresent()) {
                 LOG.warn("Unsupported operation implementation for " + op.getDescription() + ": no relationship template");
-                return Optional.absent();
+                return null;
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java:652]: Why change this?

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java [hunk 2]
```diff
@@ -152,4 +123,7 @@
      * @return the resolved property specified by key
      */
     Object resolveProperty(String nodeId, A toscaApplication, String key);
+
+    Optional<Object> getRelationshipScript(String opKey, A toscaApplication, ToscaApplication.Relationship relationship, String computeName, String expandedFolder);
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java:131]: Please add Javadoc to the new methods.

**Comment by sjcorbett** [comment]
A couple of comments, otherwise looks good.

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java (hunk 2 above)
Comment (nakomis) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java:131]: We did discuss that, but it kinda seems pointless to put javadoc to say "opKey: the key name of the operation", "toscaApplication: the tosca application". There's not really anything more that can meaningfully be added

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java (hunk 1 above)
Comment (nakomis) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java:652]: The calling code isn't expecting an Optional, so null makes (slightly) more sense

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java (hunk 2 above)
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java:131]: I'm more interested in the return value. And what is expandedFolder?

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java [hunk 3]
```diff
@@ -628,46 +593,37 @@ Operation getOperation() {
...
             }
             this.relationshipTemplate = optionalRelationshipTemplate.get();
 
             String script = getScript(artifact);
-            return buildExportStatements(op, script).or(script);
+            return Optional.of(buildExportStatements(op, script).or(script));
```
Comment (robertgmoss) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java]: Optional of Optional should just be `buildExportStatements(op, script).or(script)`

**Comment by nakomis** [comment]
PR comments addressed, additional fix added

**Comment by sjcorbett** [comment]
@nakomis Could you rebase?

Travis is also failing:

```
testConcatFunctionInTopology(io.cloudsoft.tosca.a4c.brooklyn.ToscaTypePlanTransformerIntegrationTest)  Time elapsed: 0.461 sec  <<< FAILURE!
java.lang.AssertionError: expected [Message: It Works!] but found [null]
    at org.testng.Assert.fail(Assert.java:94)
    at org.testng.Assert.failNotEquals(Assert.java:494)
    at org.testng.Assert.assertEquals(Assert.java:123)
    at org.testng.Assert.assertEquals(Assert.java:176)
    at org.testng.Assert.assertEquals(Assert.java:186)
    at io.cloudsoft.tosca.a4c.brooklyn.ToscaTypePlanTransformerIntegrationTest.testConcatFunctionInTopology(ToscaTypePlanTransformerIntegrationTest.java:384)
Results :
Failed tests: 
  ToscaTypePlanTransformerIntegrationTest>AbstractTestNGSpringContextTests.run:171->testConcatFunctionInTopology:384 expected [Message: It Works!] but found [null]
```

**Comment by nakomis** [comment]
Rebased, PR comments addressed

**Pull request merged**